projectional/src/main/java/jetbrains/jetpad/projectional/cell/BaseProjectionalSynchronizer.java

```diff
@@ -173,6 +179,30 @@ protected boolean isMultiItemPasteSupported() {
   public void setClipboardParameters(ContentKind<SourceItemT> kind, Function<SourceItemT, SourceItemT> cloner) {
     myItemKind = kind;
     myCloner = cloner;
+    myCompatibleContentKinds.put(kind, new ContentToItemMapper<>(Functions.<SourceItemT>identity(), Functions.<SourceItemT>identity()));
+  }
+
+  @Override
+  public <ContentT> void supportContentKind(ContentKind<ContentT> kind,
+      Function<ContentT, SourceItemT> fromKind,
+      Function<SourceItemT, ContentT> toKind) {
+    if (myCompatibleContentKinds.containsKey(kind)) {
+      throw new IllegalArgumentException(kind + " already supports " + myItemKind);
```

Comment:
compatibleContentKind is something that can be mapped to and from myItemKind. If some kind is already added to compatibleContentKinds, it's not OK to add it once more. Agree that message is not clear.